src/tree/v2/createTreeView.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { isAzExtTreeItem } from "@microsoft/vscode-azext-utils";
+import { TreeView, TreeViewOptions, window } from "vscode";
+import { BranchDataItemCache } from "./BranchDataItemCache";
+import { ResourceGroupsItem } from "./ResourceGroupsItem";
+import { ResourceTreeDataProviderBase } from "./ResourceTreeDataProviderBase";
+
+export interface InternalTreeView extends TreeView<ResourceGroupsItem> {
+    _reveal: TreeView<ResourceGroupsItem>['reveal'];
+}
+
+interface InternalTreeViewOptions extends TreeViewOptions<ResourceGroupsItem> {
+    treeDataProvider: ResourceTreeDataProviderBase;
+    itemCache: BranchDataItemCache;
+    /**
+     * See {@link TreeView.description}
+     */
+    description?: string;
+}
+
+/**
+ * Wrapper for `window.createTreeView`
+ * - sets the `description` if present in options
+ * - modifies `TreeView.reveal` {@link ResourceTreeDataProviderBase.reveal}
+ */
+export function createTreeView(viewId: string, options: InternalTreeViewOptions): TreeView<ResourceGroupsItem> {
+    const treeView = window.createTreeView(viewId, options);
+    treeView.description = options.description;
+
+    modifyReveal(treeView, options.treeDataProvider, options.itemCache);
+
+    return treeView;
+}
+
+function modifyReveal(treeView: TreeView<ResourceGroupsItem>, treeDataProvider: ResourceTreeDataProviderBase, itemCache: BranchDataItemCache): void {
+    (treeView as InternalTreeView)._reveal = treeView.reveal.bind(treeView) as typeof treeView.reveal;
+
+    treeView.reveal = async (element, options) => {
```

Comment:
I thought none of the v1.5 extensions called reveal? Or am I thinking of something else? If they aren't using it then we shouldn't need to monkeypatch a solution to deal with wrapping/unwrapping/compat.

Reply:
The v1.5 extensions do call reveal. There was discussion about removing all the calls, but we never went through with it.